api/agent/evictor.go

```diff
@@ -71,39 +85,42 @@ func (tok *EvictToken) isEligible() bool {
 	return true
 }
 
-func (e *evictor) GetEvictor(id, slotId string, mem, cpu uint64) *EvictToken {
+func (e *evictor) CreateEvictToken(slotId string, mem, cpu uint64) *EvictToken {
+
 	key := tokenKey{
-		id:     id,
+		id:     id.New().String(),
 		slotId: slotId,
 		memory: mem,
 		cpu:    cpu,
 	}
 
-	return &EvictToken{
-		key: key,
-		C:   make(chan struct{}),
+	token := &EvictToken{
+		key:      key,
+		C:        make(chan struct{}),
+		DoneChan: make(chan struct{}),
 	}
-}
 
-func (e *evictor) RegisterEvictor(token *EvictToken) {
-	if !token.isEligible() || token.isEvicted() {
-		return
+	if !token.isEligible() {
+		return token
 	}
 
 	e.lock.Lock()
 
-	// be paranoid, do not register if it's already there
 	_, ok := e.tokens[token.key.id]
-	if !ok {
-		e.tokens[token.key.id] = token
-		e.slots = append(e.slots, token.key)
+	if ok {
+		logrus.Fatalf("id collusion key=%+v", key)
```

Comment:
panic("") may be better... which will halt the goroutine and 503 instead of shut the server down

Reply:
there are a few other Fatalf I'd say we should probably do the same (or better, plumb the error, but if you're c++ programmer, used to making assertions, is ok life gets better...)